ctapipe/image/tests/test_image_processor.py

```diff
@@ -0,0 +1,32 @@
+from ctapipe.calib import CameraCalibrator
+from ctapipe.image import ImageProcessor
+from ctapipe.image.cleaning import MARSImageCleaner
+from numpy import isfinite
+
+
+def test_image_processor(example_event, example_subarray):
+    """ ensure we get parameters out when we input an event with images """
+
+    calibrate = CameraCalibrator(subarray=example_subarray)
+    process_images = ImageProcessor(
+        subarray=example_subarray,
+        is_simulation=True,
+        image_cleaner_type="MARSImageCleaner",
+    )
+
+    assert isinstance(process_images.clean, MARSImageCleaner)
+
+    calibrate(example_event)
+    process_images(example_event)
+
+    for dl1tel in example_event.dl1.tel.values():
+        assert isfinite(dl1tel.image_mask.sum())
```

Comment:
wouldn't be easier (and safer) to assert in a cycle over all the Field keys?

Reply:
probably - but this isn't meant to be a full test.  We already test that `hillas_parameters` and other similar functions generate the correct `Containers` as output.  Here I just want to make sure that each parameter container in the event structure is filled with non-default values (so not NaN), so I only really need to test one field from each (and assume the individual tests already cover the details).   basically the test here is "did hillas_parameters get called and store some output", not "did it fill all columns"